src/services/railgun/core/load-provider.ts

```diff
@@ -223,11 +217,20 @@ export const resumeIsolatedPollingProviderForNetwork = (
   pauseAllPollingProviders(
     networkName, // excludeNetworkName
   );
-  const pollingProviderForNetwork = pollingProviderMap[networkName];
-  if (
-    isDefined(pollingProviderForNetwork) &&
-    pollingProviderForNetwork.paused
-  ) {
-    pollingProviderForNetwork.resume();
+  const provider = providerMap[networkName];
+
+  // Check if provider exists
+  if (!isDefined(provider)) {
+    throw new Error(`No provider found for network: ${networkName}`);
   }
+
+  // Ensure provider has resume functionality 
+  if (!('paused' in provider && 'resume' in provider)) {
```

Comment:
We could do a validateProvider function rather than having this statement twice, since the flow itself its pretty similar 

something like 

```
type PausableProvider extends Provider {
 pause(): voidl; 
resume(): void;
paused: boolean;  
}


function isPausableProvider(provider: Provider): provider is PausableProvider {
  return 'pause' in provider && 'resume' in provider && 'paused' in provider;
}


const validateProvider = (networkName: NetworkName, provider?: Provider) => {
  if (!isDefined(provider)) {
    throw new Error(`No provider found for network: ${networkName}`);
  }
  if (!isPausableProvider(provider)) {
    throw new Error(`Provider for network ${networkName} is not a pausable provider`);
  }
  return provider;
};
```